The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The pre-commit check is a mandatory check, please fix detected issues.

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/qos/test_buffer.py:936:121: E501 line too long (122 > 120 characters)<br/><br/>flake8...............................................(no files to check)Skipped<br/>check conditional mark sort..........................(no files to check)Skipped<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

--- a/tests/qos/test_buffer.py
+++ b/tests/qos/test_buffer.py
@@ -933,10 +933,17 @@ def port_to_test(request, duthost):
             "LAG member port {} can not be used for dynamic buffer test".format(PORT_TO_TEST))
         PORT_TO_TEST = None
     if not PORT_TO_TEST:
-        PORT_TO_TEST = list(testPort)[0]
-    lanes = duthost.shell(
-        'redis-cli -n 4 hget "PORT|{}" lanes'.format(PORT_TO_TEST))['stdout']
-    NUMBER_OF_LANES = len(lanes.split(','))
+        # The NVIDIA SPC1 platform requires a 4 lanes port for testing to avoid exceeding the maximum available headroom
+        if duthost.facts['asic_type'].lower() == 'mellanox' and 'sn2' in duthost.facts['hwsku'].lower():
+            for port in list(testPort):
+                if duthost.count_portlanes(port) >= 4:
+                    PORT_TO_TEST = port
+                    break
+            pytest_require(PORT_TO_TEST is not None, "No 4 lanes port in DUT for Mellanox SPC1 platform")
+        else:
+            PORT_TO_TEST = list(testPort)[0]
+
+    NUMBER_OF_LANES = duthost.count_portlanes(PORT_TO_TEST)
 
     logging.info("Port to test {}, number of lanes {}".format(
         PORT_TO_TEST, NUMBER_OF_LANES))

